Source/bmalloc/libpas/src/libpas/pas_malloc_stack_logging.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "pas_config.h"
+
+#if LIBPAS_ENABLED
+
+#include "bmalloc_heap_config.h"
+#include "bmalloc_heap_inlines.h"
+#include "pas_darwin_spi.h"
+#include "pas_malloc_stack_logging.h"
+#include <stdlib.h>
+
+PAS_BEGIN_EXTERN_C;
+
+pas_msl_is_enabled_flag pas_msl_is_enabled_flag_value = pas_msl_is_enabled_flag_indeterminate;
```

Comment:
Can’t this simply be a `bool`? No code needs to read it before `compute_msl_status` status is called, so doesn’t seem like we need an indeterminate state.

Reply:
We would like to avoid calling into `pas_compute_msl_is_enabled` function if possible, and reading this from inlined function can avoid it in almost all cases.

Reply:
Is that double checked locking? Is it safe? https://en.wikipedia.org/wiki/Double-checked_locking

Reply:
@darinadler yes, this is safe, since pthread_once emits memory barrier. And we are always going to this function call if it is indeterminate